test/extended/cli/adm_upgrade/recommend.go

```diff
@@ -0,0 +1,291 @@
+package adm_upgrade
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/blang/semver/v4"
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/image"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[Serial][sig-cli] oc adm upgrade recommend", g.Ordered, func() {
```

Comment:
It has to be `Serial` because the upstream OSUS is deployed different, and then we cannot run the tests in parallel, right?

Reply:
Of course.